yuyo/__main__.py

```diff
@@ -0,0 +1,709 @@
+            id=hikari.UNDEFINED if self.id is None else hikari.Snowflake(self.id),
+            default_member_permissions=default_member_permissions,
+            is_dm_enabled=self.is_dm_enabled,
+            is_nsfw=self.is_nsfw,
+            name_localizations=self.name.localisations,
+            description_localizations=self.description.localisations,
+            options=[opt.to_option() for opt in self.options],
+        )
+
+
+class _DeclareMenuCmdModel(_CommandModel):
+    type: typing.Annotated[  # noqa: VNE003
+        typing.Literal[hikari.CommandType.MESSAGE, hikari.CommandType.USER], pydantic.PlainSerializer(_to_int)
+    ]
+    name: _MaybeLocalised
+    id: typing.Optional[_Snowflake] = None  # noqa: VNE003
+    default_member_permissions: typing.Optional[int] = None
+    is_dm_enabled: bool = True
+    is_nsfw: bool = False
+
+    @classmethod
+    def from_builder(cls, builder: hikari.api.ContextMenuCommandBuilder, /) -> Self:
+        name = _MaybeLocalised("name", builder.name, builder.name_localizations)
+
+        default_member_permissions = None
+        if builder.default_member_permissions is not hikari.UNDEFINED:
+            default_member_permissions = builder.default_member_permissions
+
+        return cls(
+            type=typing.cast("typing.Literal[hikari.CommandType.MESSAGE, hikari.CommandType.USER]", builder.type),
+            name=name,
+            id=None if builder.id is hikari.UNDEFINED else builder.id,
+            default_member_permissions=default_member_permissions,
+            is_dm_enabled=True if builder.is_dm_enabled is hikari.UNDEFINED else builder.is_dm_enabled,
+            is_nsfw=False if builder.is_nsfw is hikari.UNDEFINED else builder.is_nsfw,
+        )
+
+    def to_builder(self) -> hikari.api.ContextMenuCommandBuilder:
+        default_member_permissions = hikari.UNDEFINED
+        if self.default_member_permissions is not None:
+            default_member_permissions = self.default_member_permissions
+
+        self.name.assert_length(1, 31)
+        return hikari.impl.ContextMenuCommandBuilder(
+            type=self.type,
+            name=self.name.value,
+            id=hikari.UNDEFINED if self.id is None else hikari.Snowflake(self.id),
+            default_member_permissions=default_member_permissions,
+            is_dm_enabled=self.is_dm_enabled,
+            is_nsfw=self.is_nsfw,
+            name_localizations=self.name.localisations,
+        )
+
+
+_CommandModelIsh = typing.Union[_DeclareSlashCmdModel, _DeclareMenuCmdModel]
+
+
+class _DeclareModel(pydantic.BaseModel):
+    commands: list[_CommandModelIsh]
+
+
+async def _declare_coro(token: str, schema: _DeclareModel) -> None:
+    app = hikari.RESTApp()
+    await app.start()
+
+    async with app.acquire(token, token_type=hikari.TokenType.BOT) as rest:
+        application = await rest.fetch_application()
+        await rest.set_application_commands(application.id, [cmd.to_builder() for cmd in schema.commands])
```

Comment:
TODO: this should diff check against the current declared commands and only redeclare if it's different